test/test_jit_autocast.py

```diff
@@ -819,6 +819,27 @@ def test_nhwc_autocast_jit_trace_model(model, x):
                 continue
             test_nhwc_autocast_jit_trace_model(self.models[i], self.inputs[i])
 
+    def test_cat_promote(self):
+        class TestModel(torch.nn.Module):
+            def __init__(self):
+                super(TestModel, self).__init__()
+            def forward(self, a, b):
+                return torch.cat([a, b], 0)
+
+        test_model = TestModel().eval()
+        with torch.cpu.amp.autocast(cache_enabled=False, dtype=torch.bfloat16), torch.no_grad():
+            a = torch.rand(24, 128, 128)
+            b = torch.rand(24, 128, 128, dtype=torch.bfloat16)
+            c = test_model(a, b)
+            traced = torch.jit.trace(test_model, (a, b))
+        traced = torch.jit.freeze(traced)
```

Comment:
Is `torch.jit.freeze` necessary here?

Reply:
No, I don't think it's necessary.

Reply:
The test case updated for both `freeze` and `unfreeze` cases.